dot/network/notifications_test.go

```diff
@@ -263,16 +264,14 @@ func Test_HandshakeTimeout(t *testing.T) {
 	nodeB.noGossip = true
 
 	// create info and handler
-	info := &notificationsProtocol{
-		protocolID:            nodeA.host.protocolID + blockAnnounceID,
-		getHandshake:          nodeA.getBlockAnnounceHandshake,
-		handshakeValidator:    nodeA.validateBlockAnnounceHandshake,
-		inboundHandshakeData:  new(sync.Map),
-		outboundHandshakeData: new(sync.Map),
+	testHandshakeDecoder := func([]byte) (Handshake, error) {
+		return nil, errors.New("unimplemented")
 	}
+	info := newNotificationsProtocol(nodeA.host.protocolID+blockAnnounceID,
+		nodeA.getBlockAnnounceHandshake, testHandshakeDecoder, nodeA.validateBlockAnnounceHandshake)
 
 	nodeB.host.h.SetStreamHandler(info.protocolID, func(stream libp2pnetwork.Stream) {
-		fmt.Println("never respond a handshake message")
+		fmt.Println("never responded to a handshake message")
```

Comment:
nit out of scope slightly; not sure how this is used in a test, but maybe change to set a boolean to detect if it's called or not for example (or close a channel etc.)? Or change it to use `t.Log` although I'm not sure we care about this message unless we debug things.

Reply:
the test is testing that the handshake times out, ie nodeA sends a handshake to nodeB but nodeB never responds. this function here is what's called when nodeB receives the handshake, so it just prints something, but happy also to have a bool check. but the test isn't really to see if nodeB gets the handshake, just that nodeA times out

Reply:
If we have nothing to assert in there (or check it ever gets called), let's just remove the print (or replace by a comment) then. One can always add a `t.Log` in there to debug the test. I'm just not too much for test logs especially when running tests in parallel (some day, some day!) where it gets messy to follow where each log comes from.

Reply:
done